server/app/tests/test_user.py

```diff
@@ -58,4 +66,151 @@ def test_change_password_new_password_is_same_as_current_password(user_service,
+    user_start = User(id=1, username="Username1", email="a@email.com", role=UserRole.user, hashed_password=hash_password("Password1"))
+    user_end = User(id=1, username="Username1", email="a@email.com", role=UserRole.admin, hashed_password=hash_password("Password1"))
+
+    with mock.patch.object(user_service, "add", return_value=user_start) as _:
+        mock_user_repo.set_role.return_value = user_end
+
+        def set_role(user: User, role: UserRole):
+            user.sqlmodel_update({"role": role})
+            return user
+        mock_user_repo.set_role.side_effect = set_role
+
+        result = user_service.add_admin(dto)
+        
+        assert result.role == UserRole.admin
+
+def test_add_admin___integration():
+
+    with TestClient(app) as client:
+    
+        def get_auth_header(data: dict | None) -> dict:
+            response = client.post("/api/v1/users/login", json=data)
+            assert response.status_code == 200
+            
+            ok_request = response.json()
+            jwt = ok_request["token"]
+            header = {
+                "Authorization": f"Bearer {jwt}"
+            }
+
+            return header
+    
+        config_file = "./volume-server-cfg/superadmin_password.txt"
+        data = {
+            "username" : "admin",
+            "password" : ""
+        }
+
+        with open(config_file, "r") as f:
+            data["password"] = old_password = f.readline()
+
+        # [1] Adding an admin before changing the super admin password
+        header = get_auth_header(data)
+
+        data = {
+            "username": "Username1",
+            "email": "email1@email.com",
+            "password": "Password1"
+        }
+        
+        response = client.post("/api/v1/users/register-admin", json=data, headers=header)
+        bad_request = response.json()
+        
+        assert response.status_code == 403
+        assert bad_request["detail"] == "Password change required"
+        
+        # [2] Changing the super admin password
+        data = {
+            "old_password": old_password,
+            "new_password": "Password1" 
+        }
+        response = client.post("/api/v1/users/password", json=data, headers=header)
+        assert response.status_code == 204
+
+        data = {
+            "username" : "admin",
+            "password" : data["new_password"]
+        }
+        header = get_auth_header(data)
+
+        # [3] Adding an admin
+        data = {
+            "username": "Username1",
+            "email": "email1@email.com",
+            "password": "Password1"
+        }
+        response = client.post("/api/v1/users/register-admin", json=data, headers=header)
+        assert response.status_code == 200
+
+        created_admin = response.json()
+        assert created_admin["username"] == data["username"]
+        assert created_admin["role"] == "admin"
+
+def test_add___integration():
```

Comment:
Add two more test cases here: duplicate username and duplicate email (checking if the API will return a status code of 400).

Reply:
I initially chose not to include these tests, as the unit tests already cover this functionality, and I wanted to focus on testing the core functionality through integration tests. However, I will add these tests as suggested.